File: app/src/main/java/com/alienstar/cyrus/advancedandroid/di/ControllerKey.java
package com.alienstar.cyrus.advancedandroid.di;

import com.bluelinelabs.conductor.Controller;

import java.lang.annotation.ElementType;
import java.lang.annotation.Target;

import dagger.MapKey;

/**
 * Created by cyrus on 3/5/18.
 */
@MapKey
@Target(ElementType.METHOD)
public @interface ControllerKey {
    Class<? extends Controller> value();
}

File: app/src/main/java/com/alienstar/cyrus/advancedandroid/di/ControllerKey.kt
package com.alienstar.cyrus.advancedandroid.di

import com.bluelinelabs.conductor.Controller

import dagger.MapKey
import kotlin.reflect.KClass

/**
 * Created by cyrus on 3/5/18.
 */
@MapKey
@Target(AnnotationTarget.FUNCTION, AnnotationTarget.PROPERTY_GETTER, AnnotationTarget.PROPERTY_SETTER)
annotation class ControllerKey(val value: KClass<out Controller>)
